frontend/packages/operator-lifecycle-manager/src/components/operator-install-page.tsx

```diff
@@ -0,0 +1,484 @@
+  const apiLink = referenceForGroupVersionKind(groupVersionFor(apiVersion).group)(
+    groupVersionFor(apiVersion).version,
+  )(kind);
+  return (
+    <Button
+      isDisabled={disabled}
+      className={className}
+      variant="primary"
+      component="a"
+      href={`/k8s/ns/${crdNamespace}/${referenceForModel(ClusterServiceVersionModel)}/${
+        obj?.metadata?.name
+      }/${apiLink}/~new`}
+    >
+      {`Create ${kind}`}
+    </Button>
+  );
+};
+
+const InstallPageCRDRequiredMessage: React.FC<InstallPageMessageProps> = ({ obj }) => {
+  if (!isCRDAvailable(obj)) {
+    return <></>;
+  }
+  const crObj = safeLoad(getCRDFromObject(obj));
+
+  const crdKind = crObj?.[0]?.kind;
+  const crdNameSpace = crObj?.[0]?.metadata?.namespace;
+  const description = obj?.metadata?.annotations?.description;
+  return (
+    <>
+      <div className="co-clusterserviceversion-box">
+        <span className="co-resource-item">
+          <ResourceLink kind={crdKind} title={crdKind} name={crdKind} namepace={crdNameSpace} />
+          <RedExclamationCircleBadge message="Required" />
+        </span>
+        <p>{description}</p>
+      </div>
+    </>
+  );
+};
+
+const InstallSucceededMessage: React.FC<InstallPageMessageProps> = ({
+  namespace,
+  csvName,
+  obj,
+}) => {
+  const crdMessage = isCRDAvailable(obj)
+    ? 'The operator has installed successfully. Create the required custom resource to be able to use this operator.'
+    : '';
+  return (
+    <>
+      <span className="co-clusterserviceversion-install__heading">
+        Installed operator - ready for use
+      </span>
+      <br />
+      <span>{crdMessage}</span>
+      <br />
+      <InstallPageCRDRequiredMessage namespace={namespace} obj={obj} />
+      <div className="co-clusterserviceversion-install__button_div">
+        {!isCRDAvailable(obj) && (
+          <Button
+            variant="primary"
+            component="a"
+            href={`/k8s/ns/${namespace}/${referenceForModel(
+              ClusterServiceVersionModel,
+            )}/${csvName}`}
+          >
+            View Operator
+          </Button>
+        )}
+        <CreateCRDButton namespace={namespace} className="" obj={obj} disabled={false} />
+        <ViewInstalledOperatorsButton namespace={namespace} className={''} />
+      </div>
+    </>
+  );
+};
+const InstallingMessage: React.FC<InstallPageMessageProps> = ({ namespace, obj }) => {
+  const reason = _.startCase((obj as any)?.status?.reason || '');
+  const message = (obj as any)?.status?.message || '';
+  const crdMessage = isCRDAvailable(obj)
+    ? 'Once the operator is installed the required custom resource will be available for creation.'
+    : '';
+  const installMessage = `The operator is being installed. This may take a few minutes. ${crdMessage}`;
+  return (
+    <>
+      <span className="co-clusterserviceversion-install__heading">Installing operator</span>
+      <br />
+      {reason !== '' && (
```

Comment:
No reason not to handle `null` and `undefined`

```suggestion
      {reason && (
```